app/models/student.rb

```diff
@@ -54,6 +54,11 @@ class Student < Role
     Settings.student_csv_order || %w[user_name section_name last_name first_name id_number email]
   ).map(&:to_sym).freeze
 
+  def can_cancel_test?(test_run_id)
```

Comment:
Since this has to with authorization, a better place for this method is inside the `StudentPolicy` class, as a helper method.